src/prepack-cli.js

```diff
@@ -454,6 +457,90 @@ fi
     return fatalErrors === 0;
   }
 
+  function generateDebugRepro(
+    sourceMaps: Array<string>,
+    originalSourceFiles: Array<{ absolute: string, relative: string }>
+  ) {
+    if (reproFilePath === undefined) return;
+
+    let reproZip = zipFactory();
+    // Copy all original sourcefiles used while Prepacking.
+    for (let file of originalSourceFiles) {
+      try {
+        // To avoid copying the "/User/name/..." version of the bundle/map/model included in originalSourceFiles
+        if (!reproFileNames.includes(file.relative)) {
+          console.log(`Zipping ${file.absolute}`);
+          let content = fs.readFileSync(file.absolute, "utf8");
+          reproZip.file(file.relative, content);
+        }
+      } catch (err) {
+        console.error(`Could not zip source file: ${err}`);
+      }
+    }
+
+    // Copy all input files
+    for (let file of reproFileNames) {
+      let content = fs.readFileSync(file, "utf8");
+      reproZip.file(path.basename(file), content);
+    }
+
+    // Copy all sourcemaps (discovered while prepacking)
+    for (let map of sourceMaps) {
+      try {
+        console.log(`Zipping ${map}`);
+        let content = fs.readFileSync(map, "utf8");
+        reproZip.file(path.basename(map), content);
+      } catch (err) {
+        console.error(`Could not zip sourcemap: ${err}`);
+      }
+    }
+
+    // Copy Prepack lib and package.json to install dependencies.
+    let yarnRuntime = "yarn";
+    let yarnCommand = ["pack", "--filename", "prepack-bundled.tgz"];
+    let yarnProcess = child_process.spawn(yarnRuntime, yarnCommand);
+
+    yarnProcess.on("exit", function(code, signal) {
+      // Because zipping the .tgz causes corruption issues when unzipping, we will
```

Comment:
What are the "corruption issues"?

Reply:
`yarn pack` creates a `.tgz` file. If we use node-fs to read that file, then use node-zip to zip that file, when we unzip the overall zip file created by node-zip, the `.tgz` is corrupted -- attempts to unzip it from the commandline fail with a "unrecognized archive format" error, and attempts to open it using GUI archive utilities produce a `.cpgz` which indicates a problem in the zipping process.

Reply:
Did you file a bug report somewhere?

Reply:
[Just did](https://github.com/daraosn/node-zip/issues/30)